Automattic/jetpack
# AI Assistant: Enable backend prompts for 50% of production sites

This is a followup to #33356.

## Proposed changes:

* Keep backend prompts enabled for beta sites
* Expand backend prompts to 50% of production sites (all sites where the blog_id is multiple of 2)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test on a site with beta extensions enabled and confirm that it uses the backend prompts:
   * To check if it's using backend prompt, notice the `jetpack-ai-client:ask-question Asking question:` log message
   * Backend prompts will have messages where `role = jetpack-ai`:

<img width="500" alt="Screenshot 2023-10-09 at 16 16 24" src="https://github.com/Automattic/jetpack/assets/6760046/90367ffe-38f2-40f6-bfc9-fdfa106eb862">

* Now test on a site with production extensions enabled:
   * if the blog ID is a multiple of 2, you will see the backend prompts (`role = jetpack-ai`)
   * if the blog ID is not a multiple of 2, you will see the client prompts, with `role = system` and `role = user` messages:

<img width="500" alt="Screenshot 2023-10-09 at 16 15 24" src="https://github.com/Automattic/jetpack/assets/6760046/b4e73b89-f564-453b-97d0-4ba10fd1eb00">

* To simplify your testing, if your testing site `blog_id` is not a multiple of 2, you can switch the `blogId % 2 === 0` on the code for `blogId % 2 === 1`, build the plugin again and test it; by doing so, you can confirm if it switches between client and backend prompts depending on the value of the `blog_id`
